<launch>

  <arg name="gui" default="true"/>

  <param name="/use_sim_time" value="true"/>
  <param name="robot_description" command="$(find xacro)/xacro.py '$(find pr2_description)/robots/pr2.urdf.xacro'"/>

  <node name="rosbag_play"
        pkg="rosbag" type="play"
        args="$(find jsk_pcl_ros)/sample/data/pr2_look_around_in_room_73b2.bag --clock --loop"/>

  <node name="sample_manager"
        pkg="nodelet" type="nodelet"
        args="manager"/>

  <group ns="/kinect_head">
    <node name="republish_rgb"
          pkg="image_transport" type="republish"
          args="compressed raw">
      <remap from="in" to="rgb/image_rect_color"/>
      <remap from="out" to="rgb/image_rect_color"/>
    </node>
    <node name="republish_depth"
          pkg="image_transport" type="republish"
          args="compressedDepth raw">
      <remap from="in" to="depth_registered/hw_registered/image_rect"/>
      <remap from="out" to="depth_registered/hw_registered/image_rect"/>
    </node>
    <node name="point_cloud_xyzrgb"
          pkg="nodelet" type="nodelet"
          args="load depth_image_proc/point_cloud_xyzrgb /sample_manager">
      <remap from="rgb/camera_info" to="rgb/camera_info"/>
      <remap from="rgb/image_rect_color" to="rgb/image_rect_color"/>
      <remap from="depth_registered/image_rect" to="depth_registered/hw_registered/image_rect"/>
    </node>
  </group>

  <node name="joint_state_static_filter"
        pkg="nodelet" type="nodelet"
        args="standalone jsk_pcl/JointStateStaticFilter">
    <remap from="~input" to="/kinect_head/depth_registered/points"/>
    <remap from="~input_joint_state" to="/joint_states"/>
    <rosparam command="load" file="$(find jsk_pcl_ros)/config/pr2_sensor_joint.yaml"/>
  </node>

  <group if="$(arg gui)">
    <node name="rviz"
          pkg="rviz" type="rviz"
          args="-d $(find jsk_pcl_ros)/sample/rviz/joint_state_static_filter.rviz"/>
  </group>

</launch>
